docs/NMOS With JPEG XS.md

```diff
@@ -30,7 +30,16 @@ This has been permitted since IS-04 v1.1.
 
 Nodes implementing [BCP-004-01][] Receiver Capabilities use the existing `constraint_sets` parameter within the `caps` object, describing combinations of frame rates, width and height, and other parameters which the receiver can support, using the parameter constraints defined in the [Capabilities register](https://specs.amwa.tv/nmos-parameter-registers/branches/main/capabilities/) of the [NMOS Parameter Registers][].
 
-If the Receiver supports streams meeting the traffic shaping and delivery timing requirements for ST 2110-22, it SHOULD use the `urn:x-nmos:cap:transport:st2110_21_sender_type` parameter constraint.
+Receivers are RECOMMENDED to use the following parameter constraints:
+
+- [Frame Width](https://specs.amwa.tv/nmos-parameter-registers/branches/main/capabilities/#frame-width)
+- [Frame Height](https://specs.amwa.tv/nmos-parameter-registers/branches/main/capabilities/#frame-width)
+- [Color Sampling](https://specs.amwa.tv/nmos-parameter-registers/branches/main/capabilities/#color-sampling)
+- [Component Depth](https://specs.amwa.tv/nmos-parameter-registers/branches/main/capabilities/#component-depth)
+- [Bits Per Pixel](https://specs.amwa.tv/nmos-parameter-registers/branches/main/capabilities/#bits-per-pixel)
```

Comment:
If using bits_per_pixel instead of bitrate I think there is a missing RECOMMENDED grain rate constraint such that a given constraint set may relate to some maximum bitrate supported by the receiver. For example a receiver having a 2.5Gb network interface has a clear restriction of incoming bitrate. If it is not recommended to express that constraint using the bitrate constraint then I think there should be a set of RECOMMENDED constraints that achieve the same objective.

Reply:
Completely agree with the user story of wanting to know whether the network interface/link has the necessary bandwidth. This concern applies to all stream types including uncompressed streams of course, so warrants a general solution. Maybe it is complicated by the fact that there can be several NMOS Receivers sharing the same network interface.

Reply:
The trouble with mixing bits-per-pixel and bit rate Flow attributes and Receiver parameter constraints is that it slightly complicates their specifications and Controller implementation if we want Controllers to match a constraint expressed one way against Flow/Sender attribute expressed the other way.

Reply:
Hey all,

I have not been part of your discussions, but I do have a comment.

I think that bits per pixel and bandwidth are different and do not perfectly overlap.

A codec can have a maximum bpp that results in a lower bandwidth than what the device can support. For example, someone’s JPEG-XS implementation can max out at 4bpp on hardware with a 1GbE pipe. For 720p video, you’d hit the max bpp but not the bandwidth and knowing this prior to connection can help a controller. Similarly, there may be a minimum bpp but not so for bandwidth.

Second, an application may want to be able to set the device's allowed bandwidth, so that they can partition for multiple streams or leave space for other traffic (USB, control, etc). It's possible the same is true for bpp, but those use cases are more nuanced. Maybe an IT pro would want to further cap bpp to avoid allowing other controllers to make connections that needlessly waste bandwidth?

Reply:
I think that all agrees well with the ad hoc discussions that have been going on around this. One other related facet that has been discussed is the difference between codestream/format/Flow bit rate and transport/stream/Sender bit rate including protocol overhead (RTP, IP, etc.). It'll be easier to pull all of these discussions together when we get a formal activity group, but there are a few notes in https://github.com/AMWA-TV/nmos-parameter-registers/pull/38 and related issues on that repo.

Reply:
I think that the bit_rate approach is more "universal" and allows a broader set of capabilities/constraints to be expressed. It would be bad to have two similar approaches being used and require conversions between the bits_per_pixel world and the bit_rate world. The approximation bit_rate' = width * height * grain_rate * bits_per_pixel without considering the RTP/UDP/IP overhead require all those elements to be part of a constraint_set to be evaluated. A bit_rate constraint by itself is totally independent of the other constraints. A bits_per_pixel constraints by itself indicates nothing useful because AFAIK the profile, level, sublevel provide the operational constraints of the codec.

Reply:
> A bits_per_pixel constraints by itself indicates nothing useful because AFAIK the profile, level, sublevel provide the operational constraints of the codec.

Other implementers have indicated this is not the case, since decoders or licences can have compression ratio (bpp) limits. And for JPEG XS, these can be different from the broad constraint indicated by sublevel. See e.g. @rbgodwin-nt https://github.com/AMWA-TV/bcp-006-01/issues/6#issuecomment-1023338440

Reply:
Will be resolved by #12 which specifies Flow `sublevel_bpp` and codestream `bit_rate` and Sender `bit_rate` including the transport overhead - and related Receiver parameter constraints.